File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # hashmap --> num : count
        count = {}
        for n in nums:
            count[n]= 1 + count.get(n, 0)

        arr= []
        for n, c in count.items():
            arr.append([c, n])
        arr.sort()

        res= []
        while len(res) < k: 
            # since array is sorted, pop will take off the largest element and append it to res array
            # [1] gets the num values because it's stored [cnt, num]
            res.append(arr.pop()[1])

        return res
        

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> previous= new HashMap<>();

        for (int i=0; i < nums.length; i++) {
            int num= nums[i];
            int diff= target-num;
            
            if (previous.containsKey(diff)) {
                return new int[] {previous.get(diff), i};
            }
            previous.put(num, i);
        }

        return new int[] {};

        
    }
}
